File: src/main/java/cc/duduhuo/git/conflict/tool/Tools.java
package cc.duduhuo.git.conflict.tool;

import java.awt.*;

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/26 21:06
 * Description:
 * Remarks:
 * =======================================================
 */
public final class Tools {
    public static String color2HexString(Color color) {
        String r, g, b;
        StringBuilder su = new StringBuilder();
        r = Integer.toHexString(color.getRed());
        g = Integer.toHexString(color.getGreen());
        b = Integer.toHexString(color.getBlue());
        r = r.length() == 1 ? "0" + r : r;
        g = g.length() == 1 ? "0" + g : g;
        b = b.length() == 1 ? "0" + b : b;
        su.append("0x");
        su.append(r);
        su.append(g);
        su.append(b);
        // return like 0x000000
        return su.toString();
    }

    public static String int2HexString(int color) {
        StringBuilder s = new StringBuilder(Integer.toHexString(color));
        int count = 6 - s.length();
        for (int i = 0; i < count; i++) {
            s.insert(0, "0");
        }
        s.insert(0, "0x");
        return s.toString();
    }
}

File: src/main/java/cc/duduhuo/git/conflict/tool/Tools.kt
package cc.duduhuo.git.conflict.tool

import java.awt.Color

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/26 21:06
 * Description:
 * Remarks:
 * =======================================================
 */
object Tools {
    fun color2HexString(color: Color): String {
        val sb = StringBuilder()
        var r = Integer.toHexString(color.red)
        var g = Integer.toHexString(color.green)
        var b = Integer.toHexString(color.blue)
        r = if (r.length == 1) "0$r" else r
        g = if (g.length == 1) "0$g" else g
        b = if (b.length == 1) "0$b" else b
        sb.append("0x")
        sb.append(r)
        sb.append(g)
        sb.append(b)
        // return like 0x000000
        return sb.toString()
    }

    fun int2HexString(color: Int): String {
        val sb = StringBuilder(Integer.toHexString(color))
        val count = 6 - sb.length
        for (i in 0 until count) {
            sb.insert(0, "0")
        }
        sb.insert(0, "0x")
        return sb.toString()
    }
}
